src/components/alerting.tsx

```diff
@@ -1874,18 +1906,61 @@ const RulesPage_: React.FC<{}> = () => {
 };
 const RulesPage = withFallback(RulesPage_);
 
-const CreateButton: React.FC<{}> = React.memo(() => {
+const CreateSilenceButton: React.FC<{}> = React.memo(() => {
   const { t } = useTranslation('public');
 
   return (
     <Link className="co-m-primary-action" to="/monitoring/silences/~new">
-      <Button variant="primary" data-test="create-silence-btn">
+      <Button data-test="create-silence-btn" variant="primary">
         {t('Create silence')}
       </Button>
     </Link>
   );
 });
 
+type ExpireAllSilencesButtonProps = {
+  setErrorMessage: (string) => void;
+};
+
+const ExpireAllSilencesButton: React.FC<ExpireAllSilencesButtonProps> = ({ setErrorMessage }) => {
+  const { t } = useTranslation('public');
+
+  const [isInProgress, , setInProgress, setNotInProgress] = useBoolean(false);
+
+  const dispatch = useDispatch();
+
+  const { selectedSilences } = React.useContext(SelectedSilencesContext);
+
+  const onClick = () => {
+    setInProgress();
+    Promise.all(
```

Comment:
should we use `allSettled` instead in the case some succeed and some fail?

Reply:
Follow up PR to change this: https://github.com/openshift/monitoring-plugin/pull/34